frontend/tests/features/wheresMyGene.test.ts

```diff
@@ -569,6 +575,227 @@ describe("Where's My Gene", () => {
       ).toBeFalsy();
     });
   });
+
+  describe("Export CSV", () => {
+    test("Download CSV and validate length", async ({ page }) => {
+      await goToPage(`${TEST_URL}${ROUTES.WHERE_IS_MY_GENE}`, page);
+
+      async function getTissueSelectorButton() {
+        return page.$(getTestID(ADD_TISSUE_ID));
+      }
+
+      async function getGeneSelectorButton() {
+        return page.$(getTestID(ADD_GENE_ID));
+      }
+
+      await clickUntilOptionsShowUp(getTissueSelectorButton, page);
+      await selectFirstOption(page);
+
+      await clickUntilOptionsShowUp(getGeneSelectorButton, page);
+      await selectFirstOption(page);
+
+      await waitForHeatmapToRender(page);
+
+      async function getDownloadButton() {
+        return page.$(getTestID(DOWNLOAD_BUTTON_ID));
+      }
+
+      await clickUntilDownloadModalShowsUp(getDownloadButton, page);
+
+      const pngCheckbox = await page.$(getTestID("png-checkbox"));
+      if (!pngCheckbox) throw Error("No PNG checkbox");
+      await pngCheckbox.click();
+
+      const csvCheckbox = await page.$(getTestID("csv-checkbox"));
+      if (!csvCheckbox) throw Error("No CSV checkbox");
+      await csvCheckbox.click();
+
+      // Start waiting for download before clicking. Note no await.
+      const downloadPromise = page.waitForEvent("download");
+
+      const downloadButton = await page.$(getTestID("dialog-download-button"));
+      if (!downloadButton) throw Error("No download button");
+      await downloadButton.click();
+
+      // Wait for download
+      const download = await downloadPromise;
+
+      const fileName = download.suggestedFilename();
+
+      expect(fileName).toBe("blood.csv");
+
+      const filePath = EXPORT_OUTPUT_DIR + fileName;
+
+      await download.saveAs(filePath);
+
+      const csvBuffer = fs.readFileSync(filePath);
+
+      // Get number of lines from file
+      const numLines = csvBuffer
+        .toString()
+        .trim()
+        .split("\n")
+        .map((row) => [row]).length;
+
+      // Parse file contents
+      const parser = parse(csvBuffer, {
+        columns: false,
+        relax_column_count: true,
+        skip_empty_lines: false,
+      });
+
+      // Make sure number of lines are the same after parsing
+      expect(parser.length).toBe(numLines);
```

Comment:
Hmm this seems to me that we're testing the parser library and browser download functionality, which might not be necessary since our code doesn't touch those areas anyway?

On the other hand should we test that CSV file actually reflects the data we receive in BE response to a certain degree, along with CSV format correctness following product specs?

Thank you!

Reply:
yeah these tests were just to validate that we were making a valid CSV and isn't currently validating any of the data inside, but I can add some basic data validation!

Reply:
ahh got it that makes sense! Then that test is still needed for sure 👍 Thanks for the clarification!

Reply:
@tihuan I talked with Alec and I added some basic validation for the shape of the data. Such as making sure the column counts are correct for each row and that the number of rows match what is expected. 

The number of data rows in the CSV should be `Number of cells * number of genes selected` for example